tests/Integration/DateTimeCastTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Integration;
+
+use DateTime;
+use MongoDB\BSON\UTCDateTime;
+use Mongolid\Tests\Integration\IntegrationTestCase;
+use Mongolid\Tests\Stubs\ExpirablePrice;
+use Mongolid\Tests\Stubs\Legacy\LegacyRecordUser;
+
+class DateTimeCastTest extends IntegrationTestCase
+{
+    public function testShouldCreateAndSavePricesWithCastedAttributes(): void
+    {
+        // Set
+        $price = new ExpirablePrice();
+        $price->value = '100.0';
+        $price->expires_at = DateTime::createFromFormat('d/m/Y', '02/10/2025');
+
+        // Actions
+        $price->save();
+
+        // Assertions
+        $this->assertInstanceOf(DateTime::class, $price->expires_at);
+        $this->assertInstanceOf(UTCDateTime::class, $price->getOriginalDocumentAttributes()['expires_at']);
+
+        $price = ExpirablePrice::first($price->_id);
+        $this->assertSame('02/10/2025', $price->expires_at->format('d/m/Y'));
+        $this->assertSame(
+            '02/10/2025',
+            $price->getOriginalDocumentAttributes()['expires_at']->toDateTime()->format('d/m/Y')
```

Comment:
Esta dando erro rodando o teste nesse horario (22:42)

```suggestion
            LocalDateTime::get($price->getOriginalDocumentAttributes()['expires_at'])->format('d/m/Y')
```